GVRf/Framework/src/org/gearvrf/plugins/widget/GVRWidgetPluginActivity.java

```diff
@@ -0,0 +1,757 @@
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        // forward events to our listeners if there are any installed
+        synchronized (mAndroidEventListeners) {
+            for (int i = 0; i < mAndroidEventListeners.size; i++) {
+                mAndroidEventListeners.get(i).onActivityResult(requestCode,
+                        resultCode, data);
+            }
+        }
+    }
+
+    /**
+     * Adds an event listener for Android specific event such as
+     * onActivityResult(...).
+     */
+    public void addAndroidEventListener(AndroidEventListener listener) {
+        synchronized (mAndroidEventListeners) {
+            mAndroidEventListeners.add(listener);
+        }
+    }
+
+    /**
+     * Removes an event listener for Android specific event such as
+     * onActivityResult(...).
+     */
+    public void removeAndroidEventListener(AndroidEventListener listener) {
+        synchronized (mAndroidEventListeners) {
+            mAndroidEventListeners.removeValue(listener, true);
+        }
+    }
+
+    @Override
+    public Context getContext() {
+        return this;
+    }
+
+    @Override
+    public Array<Runnable> getRunnables() {
+        return mRunnables;
+    }
+
+    @Override
+    public Array<Runnable> getExecutedRunnables() {
+        return mExecutedRunnables;
+    }
+
+    @Override
+    public Array<LifecycleListener> getLifecycleListeners() {
+        return mLifecycleListeners;
+    }
+
+    @Override
+    public Window getApplicationWindow() {
+        return this.getWindow();
+    }
+
+    @Override
+    public Handler getHandler() {
+        return this.mHandler;
+    }
+
+    @Override
+    public void init(Objects... args) {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void destroy() {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void onCreateCallback(Bundle savedInstanceState) {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void onResumeCallback() {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void onPauseCallback() {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void onDestroyCallback() {
+        // TODO Auto-generated method stub
+
+    }
+
+    public View getWidgetView() {
+        return mWidgetView;
+    }
+
+    public int getWidth() {
+        return mViewWidth;
+    }
+
+    public int getHeight() {
+        return mViewHeight;
+    }
+
+    @Override
+    public boolean isInitialised() {
+        return mWidget.isInitialised();
+    }
+
+    public GVRScript getScript() {
+        return mScript;
+    }
+
+    public int getTextureId() {
+        return mWidget.getTexId();
+    }
+
+    public void setCurrentScript(GVRScript script) {
```

Comment:
it sets the script in base plugin activity from app, similar to setscript which does it for GVRActivity.